src/main/java/projectbookstore/service/impl/BookServiceImpl.java

```diff
@@ -38,4 +38,25 @@ public List<BookDto> findAll() {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        Book updatingBook = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException(
+                String.format("Can`t find a book with id: %d", id)
+        ));
+        updatingBook.setTitle(requestDto.getTitle());
+        updatingBook.setAuthor(requestDto.getAuthor());
+        updatingBook.setPrice(requestDto.getPrice());
+        updatingBook.setDescription(requestDto.getDescription());
+        updatingBook.setCoverImage(requestDto.getCoverImage());
+        updatingBook.setId(id);
+
+        return bookMapper.toDto(bookRepository.save(updatingBook));
```

Comment:
it can be a bit simpler than that:
```suggestion
        var updatedBook = bookMapper.toModel(requestDto);
        updatedBook.setId(id);
        return bookMapper.toDto(repo.save(updatedBook));
```
because hibernate will know that we need to update the fields of the existing book by the fact that we have set an id of an already existing book